### ISSUE TITLE
False positives selecting system allocator

### ISSUE THREAD

**Issue by birkenfeld** [rust-lang-nursery/rust-clippy#2373]
The current most recent (I think) unstable way to select the system allocator is:
```
#![feature(alloc_system, global_allocator, allocator_api)]
extern crate alloc_system;
#[global_allocator]
static A: alloc_system::System = alloc_system::System;
```
It produces the following warnings that are nonsensical in the given context:
```
warning: `0 as *mut _` detected. Consider using `ptr::null_mut()`
  --> src/main.rs:28:1
   |
28 | static A: alloc_system::System = alloc_system::System;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: #[warn(zero_ptr)] on by default
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.180/index.html#zero_ptr

warning: The function/method `alloc::heap::Alloc::usable_size` doesn't need a mutable reference
  --> src/main.rs:28:1
   |
28 | static A: alloc_system::System = alloc_system::System;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: #[warn(unnecessary_mut_passed)] on by default
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.180/index.html#unnecessary_mut_passed
```
If this select-the-allocator API is to be stabilized eventually, these should be fixed.

**Comment by oli-obk** [comment]
Probably just needs a macro check.

**Comment by phansch** [comment]
Triage: Doesn't seem to trigger with custom allocators anymore, but both lints still need an external macro check.

If someone wants to work on this, I'm happy to help out.

**Comment by samueltardieu** [comment]
Reproducer is not usable anymore. Should we close the issue?
